### ISSUE TITLE
Task runner doesn't show up in the context menu nor in its window

### ISSUE THREAD

**Issue by SelaOAL** [madskristensen/NpmTaskRunner#43]
### Installed product versions
- Visual Studio: 2015 update 3
- This extension: latest

### Description
I don't see the task runner explorer when right clicking the package.json.

### Steps to recreate
Right click it and it isn't there.
Also if I get to the task runner window from view>other windows, nothing is there.
I think it isn't being bound to the project?

**Comment by scottaddie** [comment]
@SelaOAL The extension currently only supports a `package.json` file located in the root of the project. See issue https://github.com/madskristensen/NpmTaskRunner/issues/37 for more info. Is your `package.json` file located in the root of the project?

**Comment by rand15934** [comment]
Yes it's in the root, I tried on this project: https://github.com/Autodesk-Forge/forge-boilers.nodejs/tree/master/5%20-%20viewer%2Bserver%2Boss%2Bderivatives

**Comment by SelaOAL** [comment]
Yes it's in the root, I tried on this project: https://github.com/Autodesk-Forge/forge-boilers.nodejs/tree/master/5%20-%20viewer%2Bserver%2Boss%2Bderivatives

**Comment by scottaddie** [comment]
Is there a ".njsproj" file that you're opening in VS 2015? If so, where is this on disk in relation to the actual project files?

**Comment by SelaOAL** [comment]
So I moved the solution and njsproj files to where the code was and now I see it in the conetext menu, but now it seems that there are errors when I run certain commands from there that don't happen if I run the same commands from the console.

(happens in the above project)

**Comment by datumgeek** [comment]
@scottaddie - please please please consider supporting package.json in sub-folders.  i am using spa cli tools and want to keep all the spa contents in a sub-folder.  the spa cli generates the package.json in the sub-folder also.

**Comment by scottaddie** [comment]
@datumgeek The VS Web Team is aware of the limitation. As stated by Mads in issue 37, the core Task Runner Explorer code needs to be updated in order for this extension to support it. Unfortunately, that code isn't open source, so I cannot fix it.

@madskristensen This is undoubtedly the most requested feature I see, particularly when building SPAs. Any updates on whether this will be fixed by the team?

**Comment by madskristensen** [comment]
@scottaddie Yes, and it's high on our backlog

**Comment by speedgoofy** [comment]
Just checking up on this feature, any good news coming down the pipeline?

**Comment by jayoungers** [comment]
Also checking in: with the smaller/frequent updates in VS2017 I would think this subfolder issue should be something that could be resolved in a minor update

**Comment by ElVisPL** [comment]
No update about it?

**Comment by lifeinchords** [comment]
This is still broken.

**Comment by ElVisPL** [comment]
If you using grunt or other similar tools I suggest you this solution: 

Create 'master' gruntfile in root (or equivalent for other tools) and run tasks from subdirectories from this master file

Fortunately for package.json there are nuget extensions that work for subdirectories

**Comment by lifeinchords** [comment]
Some thoughts:

I had this working when I first installed it yesterday. Today I resumed after checking in `package.json` and context menu item was gone. 

I noticed this when I set up a gulp file, which showed up in Task Runner Explorer window, but didn't see the npm stuff. Looked at the Output tab, I realized I had 2 errors.

- a syntax error in `package.json`, which I fixed
- gulpfile wasn't actually compiling: https://github.com/sindresorhus/gulp-imagemin/issues/178 which I fixed by pointing to latest node in `Program Files`
While I was there, I noticed something about Web Extensions, so I unchecked those too:

![screen shot 2017-11-21 at 5 36 14 pm](https://user-images.githubusercontent.com/716168/33100551-cdb5d386-cee2-11e7-9a8d-a67442b8bc76.png)

After that, I did this sequence, best I remember:

- Disabled NPM Task Runner extension
- Deleted `node_modules` in project root
- Copied contents of `package.json` to clipboard
- Deleted `package.json`
- Ran `npm init` from command line, which create a new `node_modules` dir in project root
- made `node_modules` is hidden (so it doesnt get checked in)
- repasted contents of `package.json`
- Set windows permissions to full control for all users for `package.json` (probably dumb but it's a dev machine and it's windows so who the hell knows permissions are the root of all evil)
- `npm install` from command line
- re-enabled NPM Task Runner extension
- restarted Studio

Then, right clicking `package.json` showed the Task Explorer item

*shrug*

**Comment by inyourspot** [comment]
I had the same issue.  Somehow the extension was disabled.  I had to go into Tools->Extension Manger and Enable it.
